sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala

```diff
@@ -469,4 +470,23 @@ class JsonFunctionsSuite extends QueryTest with SharedSQLContext {
 
     checkAnswer(sql("""select json[0] from jsonTable"""), Seq(Row(null)))
   }
+
+  test("from_json invalid json - check modes") {
+    val df = Seq("""{"a" 1}""", """{"a": 2}""").toDS()
+    val schema = new StructType().add("a", IntegerType)
+
+    checkAnswer(
+      df.select(from_json($"value", schema, Map("mode" -> "PERMISSIVE"))),
+      Row(Row(null)) :: Row(Row(2)) :: Nil)
+
+    val exceptionOne = intercept[SparkException] {
+      df.select(from_json($"value", schema, Map("mode" -> "FAILFAST"))).collect()
+    }.getMessage
+    assert(exceptionOne.contains(
+      "Malformed records are detected in record parsing. Parse Mode: FAILFAST."))
+
+    checkAnswer(
+      df.select(from_json($"value", schema, Map("mode" -> "DROPMALFORMED"))),
+      Row(null) :: Row(Row(2)) :: Nil)
```

Comment:
How does it work for DROPMALFORMED mode? This doesn't actually drop the record like JSON datasource.

Reply:
The `DROPMALFORMED` mode returns `null` for malformed JSON lines. User can filter them out later. @HyukjinKwon Do you know how to drop rows in `UnaryExpression`s?

Reply:
Nope, only possibility I raised was to make it generator expression. I haven't proposed a parse mode for this reason so far.